src/EventListener/WriteListener.php

```diff
@@ -62,12 +63,15 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
 
                 $event->setControllerResult($persistResult ?? $controllerResult);
 
-                // Controller result must be immutable for _api_write_item_iri
-                // if it's class changed compared to the base class let's avoid calling the IriConverter
-                // especially that the Output class could be a DTO that's not referencing any route
-                if (null !== $this->iriConverter && (false !== $attributes['output_class'] ?? null) && $attributes['resource_class'] === ($class = \get_class($controllerResult)) && $class === \get_class($event->getControllerResult())) {
+                if (null === $this->iriConverter || false === $attributes['output_class']) {
+                    break;
+                }
+
+                try {
                     $request->attributes->set('_api_write_item_iri', $this->iriConverter->getIriFromItem($controllerResult));
+                } catch (InvalidArgumentException $e) {
```

Comment:
Throwing an exception on the hot path should be avoided

Reply:
it would be a DTO/Resource without an iri i guess? what do you suggest?